compiler/dialects/acc.py

```diff
@@ -189,12 +212,98 @@ def verify_(self) -> None:
             )
 
 
+class AcceleratorSymbolOpTrait(SymbolOpInterface):
+    def get_sym_attr_name(self, op: Operation) -> StringAttr | None:
+        assert isinstance(op, AcceleratorOp)
+        return StringAttr(op.name_prop.string_value())
+
+
+@irdl_op_definition
+class AcceleratorOp(IRDLOperation):
```

Comment:
The accelerator ops would then not be inserted by the compiler based on the operations in the file, but rather in a (separate) target_configuration, which includes a trait for every accelerator in the system.

Reply:
Not sure what you mean by global trait. Do you mean an attribute alias? (similar to the way you handle the strides for the affine maps?)

I guess it could be done that way too, having a (verified) unique operation just encodes better (imo) that a given accelerator always has the same fields. If you attach this info via attributes you enable mistakes like different attributes referring to different fields.